### ISSUE TITLE
Ranlanet S3DIS inference on custom ply data

### ISSUE THREAD

**Issue by Parikshit00** [intel-isl/Open3D-ML#270]
Hello,
I was using Randlanet pretrained on s3dis dataset for semantic segmentation. The results are good on test sets. how can I predict unlabelled ply files using the  pretrained randlanet. I cant find my way around this. my code:  
```python
#!/usr/bin/env python
import open3d.ml.torch as ml3d
import argparse
import math
import numpy as np
import os
import random
import sys
import torch
from plyfile import PlyData, PlyElement
from os.path import exists, join, isfile, dirname, abspath, split
import open3d.ml as _ml3d



def get_custom_data(pc_names, path):
    pc_data = []
    for i, name in enumerate(pc_names):
        pc_path = join(path, name + '.ply')

        data = PlyData.read(pc_path)['vertex']

        point = np.zeros((data['x'].shape[0], 3), dtype=np.float32)
        point[:, 0] = data['x']
        point[:, 1] = data['y']
        point[:, 2] = data['z']

        feat = np.zeros(point.shape, dtype=np.float32)
        feat[:, 0] = data['red']
        feat[:, 1] = data['green']
        feat[:, 2] = data['blue']

        data = {
            'point': point,
            'feat': feat,
            'label': np.zeros([point.shape[0]])            
        }
        pc_data.append(data)

    return pc_data

s3dis_labels = {
    0: 'unlabeled',
    1: 'ceiling',
    2: 'floor',
    3: 'wall',
    4: 'beam',
    5: 'column',
    6: 'window',
    7: 'door',
    8: 'table',
    9: 'chair',
    10: 'sofa',
    11: 'bookcase',
    12: 'board',
    13: 'clutter'
}


def pred_custom_data(pc_names, pcs, pipeline_r):
    vis_points = []
    for i, data in enumerate(pcs):
        name = pc_names[i]

        results_r = pipeline_r.run_inference(data)
        pred_label_r = (results_r['predict_labels'] + 1).astype(np.int32)
        pred_label_r[0] = 0

        pts = data['point']
        feature = data['feat']
        label = data['label']

        vis_d = {
            "name": name,
            "points": pts,
            "labels": label,
            "features": feature,
            "pred": pred_label_r,
        }
        vis_points.append(vis_d)

    return vis_points

def main():
    v = ml3d.vis.Visualizer()
    lut = ml3d.vis.LabelLUT()
    for val in sorted(s3dis_labels.keys()):
        lut = ml3d.vis.LabelLUT()
        lut.add_label(s3dis_labels[val], val)
    v.set_lut("pred", lut)
    v.set_lut("labels", lut)

    randlanet_url = "https://storage.googleapis.com/open3d-releases/model-zoo/randlanet_s3dis_202010091238.pth"

    ckpt_folder = "./logs"
    ckpt_path = ckpt_folder + "/vis_weights_{}.pth".format('RandLANet')
    if not exists(ckpt_path):
        cmd = "wget {} -O {}".format(randlanet_url, ckpt_path)
        os.system(cmd)
    cfg_file = "randlanet_s3dis.yml"
    cfg = _ml3d.utils.Config.load_from_file(cfg_file)
    cfg.model.ckpt_path = ckpt_path

    model = ml3d.models.RandLANet(**cfg.model)
    pipeline = ml3d.pipelines.SemanticSegmentation(model, **cfg.pipeline)
    pipeline.load_ckpt(model.cfg.ckpt_path)



    data_path = "/home/mr-one/Desktop/Open3d-Train/dataset/test"
    pc_names = ["test"]
    pcs = get_custom_data(pc_names, data_path)

    pcs_with_pred = pred_custom_data(pc_names, pcs, pipeline)

    v.visualize(pcs_with_pred) 


if __name__ == "__main__":
    main()

```

**Comment by Parikshit00** [comment]
UPDATE:  
I just changed the `feat` parameter in get_custom_data() function and now the code seems to work fine. Predictions are fairly good. The model can detect ceilings, walls and floors very accurately. However, the model (`randlanet_s3dis_202010091238.pth`) finds it problematic to detect some classes (mainly sofa and doors which are predicted as 'clutters'). The model `randlanet_s3dis_area5_202010091333utc.pth` has really poor accuracy and needs major improvements.  

Incase someone is stuck in inferring ply files using randlanet s3dis, just assign feature attribute in dictionary `data` in `get_custom_data()` function as length of vertex points in ply file:  
```
        data = {
            'point': point,
            'feat': feat,
            'label': np.zeros(len(point))            
        }
``` 

My output on indoor bedroom ply file:  
![Screenshot from 2021-05-11 12-29-07](https://user-images.githubusercontent.com/39686233/117770490-b34b8500-b254-11eb-9dc4-6743f8e4755a.png)  
Thank you for Open3d and Open3d Ml library.
